components/esp_websocket_client/esp_websocket_client.c

```diff
@@ -730,13 +758,23 @@ static void esp_websocket_client_task(void *pv)
                     client->run = false;
                     break;
                 }
+
+                esp_websocket_client_dispatch_event(client, WEBSOCKET_EVENT_BEFORE_CONNECT, NULL, 0);
+
                 int result = esp_transport_connect(client->transport,
                                                    client->config->host,
                                                    client->config->port,
                                                    client->config->network_timeout_ms);
                 if (result < 0) {
                     ESP_LOGE(TAG, "Error transport connect %i", result);
-                    esp_websocket_client_error_connection(client);
+
+                    int status = esp_transport_ws_get_http_status_code(client->transport);
+                    if (status > 0) {
```

Comment:
Since I could not find `esp_transport_ws_get_http_status_code`, I cannot say if below code is correct.

Reply:
Note the link in the PR description. Related IDF PR: https://github.com/espressif/esp-idf/pull/9338

Reply:
Thank you for the link but PR: https://github.com/espressif/esp-idf/pull/9338 goes to v4.4.  `esp_transport_ws_get_http_status_code` needs to be merged somewhere to `master` as well.

Reply:
https://github.com/espressif/esp-idf/pull/9988 is the IDF 5.0 PR